File: csharp/isogram/Isogram.cs
using System;

public static class Isogram
{
    public static bool IsIsogram(string word)
    {
        string modified = word.ToLower().Replace(" ", "").Replace("-", "");
        for (int i = 0; i < modified.Length - 1; i++)
        {
            if (Count(modified, modified[i], i) > 1){
                return false;
            }
        }
        return true;
    }

    public static int Count(string word, char toFind, int index)
    {
        int count = 1;
        for (int i = index + 1; i < word.Length; i++)
        {
            if (word[i] == toFind){
                count++;
            }
        }
        return count;
    }
}

File: cpp/isogram/isogram.cpp
#include "isogram.h"
#include <unordered_set>

namespace isogram
{

    bool is_isogram(const std::string &phrase)
    {
        std::unordered_set<char> seen;

        for (char c : phrase)
        {
            if (c == ' ' || c == '-')
            {
                continue;
            }

            char letter = static_cast<char>(std::tolower(static_cast<unsigned char>(c)));

            if (seen.count(letter))
            {
                return false;
            }

            seen.insert(letter);
        }

        return true;
    }

} // namespace isogram
